### PR TITLE
Add rescan_for property to the metafile schema

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1395

### Change description ###
- Add `rescan_for` property to the metafile schema
- `rescan_for` is not a mandatory field. 
- When the value is present, the file name format should match the existing property `zip_file_name`.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BlobProcessorTest.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public void should_process_zipfile_after_upload_and_set_status() {
     }
 
     @Test
-    public void should_process_zipfile_with_supplementary_evidence_with_oce_classification() {
+    public void should_process_zipfile_with_supplementary_evidence_with_ocr_classification() {
```
Comment (aliveni) [inline src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/BlobProcessorTest.java:39]: not related to the change, but noticed when adding the rescan_for property to the existing functional test data

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java [hunk 2]
```diff
@@ -258,6 +258,22 @@ public void should_not_parse_envelope_with_duplicate_payment_dcn_numbers() throw
             .hasMessageContaining("instance: {\"pointer\":\"/payments\"}");
     }
 
+    @Test
+    public void should_not_parse_envelope_with_invalid_zip_file_name_in_rescan_for_property() throws IOException {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java:262]: will metadata with entry like `"rescan_for" : null` pass validation?

File: src/main/resources/metafile-schema.json [hunk 3]
```diff
@@ -48,6 +48,12 @@
       "title": "The name of the zip file",
       "pattern": "^\\d+_([012][0-9]|30|31)-([0][0-9]|[1][012])-[2][0][0-9][0-9]-([01][0-9]|[2][0123])-[0-5][0-9]-[0-5][0-9]\\.(test\\.)?zip$"
     },
+    "rescan_for": {
+      "id": "#/properties/rescan_for",
+      "type": "string",
```
Comment (doncem) [inline src/main/resources/metafile-schema.json:53]: can it be null?

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java (hunk 2 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/MetafileJsonValidatorTestForInvalidFiles.java:262]: let's add an extra test to verify it passes validation, when field is not included (similar to `should_parse_envelope_data_with_no_case_number`)

File: src/main/resources/metafile-schema.json (hunk 3 above)
Comment (rkondratowicz) [inline src/main/resources/metafile-schema.json:53]: I think this should be `"type": ["string", "null"],`

**Review by rkondratowicz — CHANGES_REQUESTED** [review]
let's add more tests

File: src/main/resources/metafile-schema.json (hunk 3 above)
Comment (aliveni) [reply src/main/resources/metafile-schema.json:53]: no, I'll update the `type` values to support null

**Comment by aliveni** [comment]
> let's add more tests

Added tests for `no rescan_for` scenario and `rescan_for` with null value

**Review by rkondratowicz — APPROVED** [review]

**Review by mdayican — APPROVED** [review]

**Pull request merged**